src/audio/module_adapter/module_adapter.c

```diff
@@ -313,6 +313,21 @@ int module_adapter_prepare(struct comp_dev *dev)
 		return -EINVAL;
 	}
 
+	/* Check if simple copy client has only one source and one sink buffer to use a
+	 * simplified copy function. The multi_source_or_sink is set later to true
+	 * if module_adapter_bind() indicates multiple sources.
+	 */
+	if (mod->num_input_buffers == 1 && mod->num_output_buffers == 1) {
+		mod->source_comp_buffer = list_first_item(&dev->bsource_list,
+							  struct comp_buffer, sink_list);
+		mod->sink_comp_buffer = sink;
+		mod->stream_copy_single_to_single = true;
+	} else {
+		mod->source_comp_buffer = NULL;
+		mod->sink_comp_buffer = NULL;
+		mod->stream_copy_single_to_single = false;
```

Comment:
`.stream_copy_1to1`?

Reply:
I was thinking 1to1 looks too much like 1:1 that would give wrong impression about doing just memcpy() from buffer to buffer.